### PR TITLE
Add `tls-server-name` to https requests if provided

### PR DESCRIPTION
**Pull request by tigrato**
This should address #887

cc @brendandburns

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -161,6 +161,10 @@ export class KubeConfig {
                 username: user.username,
             };
         }
+
+        if (cluster && cluster.tlsServerName) {
+            opts.agentOptions = { servername: cluster.tlsServerName } as https.RequestOptions;
```
Comment (brendandburns) [inline src/config.ts:166]: I think this should just be `opts.servername = cluster.tlsServerName` instead of `opts.agentOptions`

**Comment by tigrato** [comment]
@brendandburns, thanks for the review.
I pushed unit tests for  `applyToRequest` and `applytoHTTPSOptions` public functions!

File: src/config.ts (hunk 1 above)
Comment (brendandburns) [reply src/config.ts:166]: Ah, looks like you're right. Apologies for the mis-direction.

**Pull request merged**